rfcs/text/0000-tech-planning-process.md

```diff
@@ -0,0 +1,136 @@
+RECOMMENDED" "MAY", and "OPTIONAL" in this document are to be interpreted as described in BCP 14, RFC2119, and RFC8174
+when, and only when, they appear in all capitals, as shown here.
+
+### User Stories
+
+As a TL of the DCP, I am able to surface important technical priorities to Product Management that will be incorporated
+into the yearly roadmap. 
+
+As a developer on the DCP, I am able to allocate enough time to work on high priority technical architectural issues
+knowing that the work will facilitate quicker development of feature requests in the long term as informed by
+the roadmap themes.
+
+## Definitions
+**Quarter:** One quarter is equal to 13 weeks. Generally the quarters have the following timeline:
+
+- Quarter 1 (Q1): first Monday of January -> last Friday of March
+- Quarter 2 (Q2): first Monday of April -> last Friday of June
+- Quarter 3 (Q3): first Monday of July -> last Friday of September
+- Quarter 4 (Q4): first Monday of October -> last Friday of December (before the holidays)
+
+**Milestone:** a four week timeline/duration in a Release (sometimes five when a holiday is involved).
+
+**Release:** composed of three Milestones and occur on a quarterly cadence.
+
+**TL Roadmapping Shepherd:** Each quarter one TL acts as the “Shepherd” to make sure the TL planning process takes place
+in time to produce the expected Technical Objectives RFC for presentation to the TL+PM planning meeting. The Shepherd
+role will rotate once a quarter according to a spreadsheet drawn up post the approval of this RFC. 
+
+**Technical Objectives RFC:** The final artifact that will be presented to the TL+PM planning call as a representation
+of what TL would like to prioritize work on in the following quarter.
+
+**Component:** Also known as a “Box” or a *chartered component*; represents a piece of the DCP led by a TL and worked on
+by a team of software engineers. The list of chartered components are maintained
+[here](https://github.com/HumanCellAtlas/dcp-community/tree/master/charters).
+
+
+## Detailed Design
+
+### Input to the TL Planning Process
+
+1) Date of TL+PM planning call (information to be gathered by the TL Roadmapping Shepherd).
+
+2) Epics/Objectives from previous quarter that will slip the release and rollover into the next.
+
+### Timeline
+
+Per the Roadmaps and Planning Process, a TL+PM call is expected to take place during the third week of Milestone 3 in a
+Release quarter.
+
+| &nbsp;&nbsp;Week in Milestone (Week in Quarter) &nbsp;&nbsp;| Deliverables |
+|           :----:                    | :--- |
+| **M2W3 (Week 7)**  | - Each component MUST enumerate the work they believe the DCP should prioritize in the following quarter, along with rough estimates of time (sized using T-shirt sizes) and specify any other activities that will also occur (e.g. onboarding new team members) and rough estimates of time. |
```

Comment:
If this process is meant for TLs to surface their thoughts about the technical requirements of the DCP, specifying "other activities" seems like a distraction.

Reply:
I put this here because I feel like one of the common reasons for overcommitment is not allocating enough time to other activities. Surfacing them at least creates some constraints when trying to consolidate things because when you see everything on the table, you realize that the math doesn't check out.

^^ last sentence somewhat stolen from Marie Kondo.